src/components/ItemSelection.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from "react"
+import { Items, itemRarities } from "../utilities/types"
+import urls from "../utilities/urls"
+
+export type AllItems = {
+  [itemRarity in itemRarities]: {
+    items: Items[]
+  }
+}
+
+const ItemSelection = () => {
+  const [allItems, setAllItems] = useState<AllItems>({
+    Common: {items: []},
+    Uncommon: {items: []},
+    Legendary: {items: []},
+  })
+
+  useEffect(() => {
+    const fetchItems = async () => {
+
+      const itemURLS: {url: string}[] = [{url: urls.commonItemsURL}, {url: urls.unCommonItemsURL}, {url: urls.legendaryItemsURL}];
```

Comment:
what's the intention behind wrapping the URLs in an object?

```
const itemURLS: string[] = [urls.commonItemsURL, urls.unCommonItemsURL, urls.legendaryItemsURL];
```

Reply:
the usage below

`const fetchPromises = itemURLS.map(async ({url}) => {`

seems like it can be replace by

`const fetchPromises = itemURLS.map(async (url: string) => {`

Reply:
updated to treat it as array of strings, no object